server/plugin.go

```diff
@@ -20,9 +21,30 @@ type Plugin struct {
 	configuration *configuration
 }
 
-// ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
 func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "Hello, world!")
-}
+	urlBytes, _ := ioutil.ReadAll(r.Body)
+	url := string(urlBytes)
+
+	if url == "" {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Fprintf(w, "Error getting response from %s\n%s", url, err)
+		w.WriteHeader(http.StatusBadGateway)
+		return
+	}
+	defer resp.Body.Close()
 
-// See https://developers.mattermost.com/extend/plugins/server/reference/
+	respBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
We should also use an `io.LimitReader` with an arbitrary limit to prevent infinitely reading from a malicious webserver. We ran into that with a bunch of the OpenGraph stuff in the past.